### DISCUSSION TITLE
Store images on SD card separate partition

### DISCUSSION THREAD

**Discussion by oferardi** [AxisCommunications/discussions#111] [ACAP] [answered]
I need that my ACAP4 application will store snapshots of image to a separate SD card partition.

Is it possible to divide the SD they will be saved in separate partition than then video recordings?

As the support for axStorage is dropped how can I do it? My acap3 product is based on these capabilites.

If it will not be supported anymore I'll have to develop and use and external microcontroller and install it inside the camera housing with a switch.

Does it makes sense? Why to drop support suddenly after that there are existing customers that developed apps based on thouses capabilities?

**Comment by shadysp** [comment, marked as answer]
There are no plans right now to discontinue ACAP3, latest version 3.5 does support latest Firmwares and chips including Ambarella cv25 and ARTPEC-8.
If you wish to use Axstorage API or any of the other ones that is not included in ACAP4, please stay with ACAP3. 

The major difference right now between the latest releases of both other than the missing APIs is the GCC version.

- SDK4.4: GCC 11.3.0
- SDK3.5: GCC 10.2.0

- ACAP3 SDK online documentation:
https://help.axis.com/acap-3-developer-guide#hardware-compatibility 

- ACAP 3.5 APIs:
https://www.axis.com/techsup/developer_doc/acap3/3.5/main/html/acap3_api.html

Yes it makes a lot of sense, ACAP4 still developing so we are in the early stage of the transition period between both, the dropped APIs will be replaced with a new ones, and we will continue maintain ACAP3 till the same functions and more are included in ACAP4. 

Please see this link explains more the difference between the two.
https://axiscommunications.github.io/acap-documentation/docs/introduction/acap4-vs-acap3.html#what-are-the-differences-between-acap-3-sdk-and-acap-4-native-sdk

**Reply by oferardi** [reply]
Hi

  Thank for the answer.

   In the documentation of acap4 it written that axStorage Will not be moved to Native SDK.

  So can I be sure that axis will support the same storage functionality with new api? I want to confirm..

This is critical for me to have future support of my product that is based on acap3

Thanks

Ofer

**Reply by shadysp** [reply]
Yes that is the goal, although I will refer to the same reply in #112.
I can add to it "take the Old Capture API(Media capture) as an example, it was not moved to the new SDK, because it was replaced with a newer and better one the libvdo or  (video capture API). 
We understand your concern and it is a valid one, this is an important function so there are no intention from our side to stop supporting it.

**Comment by keesschollaart81** [comment]
@mattias-kindborg-at-work when using acap4, is it 'OK' to write to `/var/spool/storage/areas/SD_DISK/<myappid>/`? It seems to work but I'm not sure looking at the way stuff seems to be `mount`ed:
```
....
tmpfs on /var/spool type tmpfs (rw,relatime)
...
/dev/mmcblk1p1 on /var/spool/storage/SD_DISK type ext4 (rw,nosuid,nodev,noexec,noatime,nodiratime,lazytime,commit=1,data=ordered)
...
``` 
It there a risk for me to interfere with the recorder? Can I fill the memory in a way that would cause instability?
